src/ucs/datastruct/ptr_array.c

```diff
@@ -200,9 +199,50 @@ unsigned ucs_ptr_array_insert(ucs_ptr_array_t *ptr_array, void *value)
     return element_index;
 }
 
+void ucs_ptr_array_set(ucs_ptr_array_t *ptr_array, unsigned element_index,
+                       void *new_val)
+{
+    ucs_ptr_array_elem_t *elem;
+    unsigned next, free_iter, free_ahead;
+
+    if (ucs_unlikely(element_index > ptr_array->size)) {
+        ucs_ptr_array_grow(ptr_array, element_index + 1
```

Comment:
maybe grow to max(ptr_array->size * 2, element_index + 1)?
to amortize cost of set(1), set(2), set(3), ...